lightcycle-integration-test/src/test/java/com/soundcloud/lightcycle/sample/basic/SupportFragmentLoggerTest.java

```diff
@@ -0,0 +1,54 @@
+package com.soundcloud.lightcycle.sample.basic;
+
+import com.soundcloud.lightcycle.sample.basic.callback.FragmentLifecycleCallback;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadows.support.v4.SupportFragmentController;
+
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 23)
+public class SupportFragmentLoggerTest {
+
```

Comment:
In general, I think it would be easier if each test method was testing one behavior. 
What do you think about something like the following ? 

```java
  public class SupportFragmentLoggerTest {
    private final SampleSupportFragment sampleFragment = new SampleSupportFragment();
    private final SupportFragmentLogger supportFragmentLogger = sampleFragment.supportFragmentLogger;
    private final SupportFragmentController controller = SupportFragmentController.of(sampleFragment);

    @Test
    public void onCreate() {
        controller.create();

        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onAttach)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onCreate)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onViewCreated)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onActivityCreated)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onStart)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onResume)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onPause)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onStop)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onDestroy)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onDetach)).isFalse();
    }

    @Test
    public void onResume() {
        controller.create();
        controller.start();
        controller.resume();

        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onAttach)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onCreate)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onViewCreated)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onActivityCreated)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onStart)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onResume)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onPause)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onStop)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onDestroy)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onDetach)).isFalse();
    }

    @Test
    public void onPause() {
        controller.create();
        controller.start();
        controller.resume();
        controller.pause();

        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onAttach)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onCreate)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onViewCreated)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onActivityCreated)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onStart)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onResume)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onPause)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onStop)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onDestroy)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onDetach)).isFalse();
    }

    @Test
    public void onStop() {
        controller.create();
        controller.start();
        controller.stop();

        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onAttach)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onCreate)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onViewCreated)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onActivityCreated)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onStart)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onResume)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onPause)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onStop)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onDestroy)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onDetach)).isFalse();
    }

    @Test
    public void onDestroy() {
        controller.create();
        controller.destroy();

        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onAttach)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onCreate)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onViewCreated)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onActivityCreated)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onStart)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onResume)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onPause)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onStop)).isFalse();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onDestroy)).isTrue();
        assertThat(supportFragmentLogger.isFragmentLifecycleCallbackCalled(FragmentLifecycleCallback.onDetach)).isTrue();
    }
}

```